api/serializers.py

```diff
@@ -27,3 +27,37 @@ class ExportInstructorLocationsSerializer(serializers.ModelSerializer):
     class Meta:
         model = Airport
         fields = ('name', 'latitude', 'longitude', 'instructors')
+
+
+class EventSerializer(serializers.ModelSerializer):
+    humandate = serializers.SerializerMethodField()
+    country = serializers.CharField()
+
+    def get_humandate(self, obj):
+        """Render start and end dates as human-readable short date."""
+        return EventSerializer.human_readable_date(obj.start, obj.end)
+
+    @staticmethod
+    def human_readable_date(date1, date2):
+        """Render start and end dates as human-readable short date."""
+        if date1 and not date2:
+            return '{:%b %d, %Y}-???'.format(date1)
+        elif date2 and not date1:
```

Comment:
This is a utility function, it should not perform any validation.

By the way, this function is now only used for past/ongoing/upcoming events, so they do have a starting date. Only unpublished may not have any dates.